src/services/BlockSynchronizer.ts

```diff
@@ -74,15 +74,42 @@ class BlockSynchronizer {
 
     const [leavesSynchronizers, blockIds] = await this.processBlocks(chainStatusNextBlockId, validators, mongoBlocks);
 
-    if (blockIds.length > 0) {
-      this.logger.info(`Synchronized leaves for blocks: ${blockIds.join(',')}`);
-      const updated = await this.updateSynchronizedBlocks(await Promise.all(leavesSynchronizers), blockIds);
-      const success = updated.updatedFinalizedBlocks;
-      const failed = updated.updatedFailedBlocks;
-      this.logger.info(`Finalized successfully/failed: ${success}/${failed}. Total: ${success + failed}`);
-    } else {
-      this.logger.info('nothings was synchronised');
+    if (!blockIds.length) {
+      return;
+    }
+
+    this.logger.info(`Synchronized leaves for blocks: ${blockIds.join(',')}`);
+
+    const results = await Promise.all(
+      leavesSynchronizers.map(async (leavesSynchronizer, i) => {
+        try {
+          const status = await leavesSynchronizer;
```

Comment:
the naming here is a bit confusing, is this a promise? sounds like a service

Reply:
I renamed to `leavesSynchronizationStatuses`. 

This is not a service, type of this variable is: `Promise<boolean | null>[]`, that's why we need to `await`.